dataframely/columns/array.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) QuantCo 2025-2025
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import math
+from collections.abc import Callable, Sequence
+from typing import Any, Literal
+
+import polars as pl
+
+from dataframely._compat import pa, sa, sa_TypeEngine
+from dataframely.random import Generator
+
+from ._base import Column
+from .struct import Struct
+
+
+class Array(Column):
+    """A fixed-shape array column."""
+
+    def __init__(
+        self,
+        inner: Column,
+        shape: int | tuple[int, ...],
+        *,
+        nullable: bool = True,
+        # polars doesn't yet support grouping by arrays,
+        # see https://github.com/pola-rs/polars/issues/22574
```

Comment:
nit: Should we maybe move this down to where we have to make the update once this is released?

```python
super().__init__(
    nullable=nullable,
    # polars doesn't yet support grouping by arrays,
    # xref: https://github.com/pola-rs/polars/issues/22574
    primary_key=False,
    check=check,
    alias=alias,
    metadata=metadata,
)
```

Reply:
I think it's fine, we need to adjust both places anyways.